src/full/Agda/TypeChecking/Rules/Data.hs

```diff
@@ -1676,8 +1676,7 @@ bindParameters
 bindParameters 0 [] a ret = ret EmptyTel a
 
 bindParameters 0 (par : _) _ _ = setCurrentRange par $
-  typeError . GenericDocError =<< do
-    text "Unexpected parameter" <+> prettyA par
+  typeError $ UnexpectedParameter par
```

Comment:
Remark: 
This error is produced in issue #555 
```agda
data Exp : Set
data Exp Γ where
```
I think this can only happen here, for `data` definitions, and analogously for `record definitions`.
```agda
record R : Set
record R A where
```
I think the `bindParameters` function is reused for both, so that covers both.